libs/fluid-elements/button/src/lib/button.ts

```diff
@@ -0,0 +1,170 @@
+ * @cssprop --fluid-button--foreground-warning-active - Controls the foreground active color for the warning setting.
+ * @cssprop --fluid-button--background-warning-active - Controls the background active color for the warning setting.
+ * @cssprop --fluid-button--foreground-negative - Controls the foreground color for the negative setting.
+ * @cssprop --fluid-button--background-negative - Controls the background color for the negative setting.
+ * @cssprop --fluid-button--foreground-negative-hover - Controls the foreground hover color for the negative setting.
+ * @cssprop --fluid-button--background-negative-hover - Controls the background hover color for the negative setting.
+ * @cssprop --fluid-button--foreground-negative-active - Controls the foreground active color for the negative setting.
+ * @cssprop --fluid-button--background-negative-active - Controls the background active color for the negative setting.
+ * @cssprop --fluid-button--foreground-disabled - Controls the foreground color for the disabled state.
+ * @cssprop --fluid-button--background-disabled - Controls the background color for the disabled state.
+ */
+export class FluidButton extends LitElement {
+  /** Styles for the button component */
+  static get styles(): CSSResult {
+    return styles;
+  }
+
+  /**
+   * Defines if the button is disabled or not.
+   * @attr
+   * @type boolean
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Defines the color theme of the button.
+   * @attr
+   * @type {'main' | 'positive' | 'warning' | 'error' }
+   * @default 'main'
+   */
+  @property({ type: String, reflect: false })
+  get color(): FluidButtonColor {
+    return this._color;
+  }
+  set color(value: FluidButtonColor) {
+    const oldColor = this._color;
+    this._color = value ? value : 'main';
+    this.requestUpdate('color', oldColor);
+  }
+  private _color: FluidButtonColor = 'main';
+
+  /**
+   * Defines the emphasis of the button.
+   * @attr
+   * @type {'low' | 'medium' | 'high'}
+   * @default 'medium'
+   */
+  @property({ type: String, reflect: false })
+  get emphasis(): FluidButtonEmphasis {
+    return this._emphasis;
+  }
+  set emphasis(value: FluidButtonEmphasis) {
+    const oldEmphasis = this._emphasis;
+    this._emphasis = value ? value : 'medium';
+    this.requestUpdate('emphasis', oldEmphasis);
+  }
+  private _emphasis: FluidButtonEmphasis = 'medium';
+
+  /**
+   * Defines the size of the button.
+   * @attr
+   * @type {'small' | 'default' | 'large' }
+   * @default 'large'
+   */
+  @property({ type: String, reflect: false })
+  get size(): FluidButtonSize {
+    return this._size;
+  }
+  set size(value: FluidButtonSize) {
+    const oldSize = this._size;
+    this._size = value ? value : 'large';
+    this.requestUpdate('size', oldSize);
+  }
+  private _size: FluidButtonSize = 'large';
+
+  constructor() {
+    super();
+  }
```

Comment:
If you don't perform anything inside the constructor it should not be needed.

```suggestion
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor

![image](https://user-images.githubusercontent.com/11156362/80590359-096bd280-8a1c-11ea-839e-f6558fc0d9ed.png)